In [2]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [32]:
players_data <- read_csv('data/players.csv')
sessions_data <- read_csv('data/sessions.csv')
#players_data

na_players <- select(players_data, experience:age) |>
    filter(if_any(everything(), is.na))
na_players
# sessions_data

Rows: 196 Columns: 9
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, age
lgl (3): subscribe, individualId, organizationName

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe hashedEmail played_hours name gender age

**Task: Description of the dataset**

Provide a full descriptive summary of the dataset, including information such as the number of observations, number of variables, name and type of variables, what the variables mean, any issues you see in the data, any other potential issues related to things you cannot directly see, how the data were collected, etc. Make sure to use bullet point lists or tables to summarize the variables in an easy-to-understand format.

Note that the selected dataset will probably contain more variables than you need. In fact, exploring how the different variables in the dataset affect your model may be a crucial part of the project. You need to summarize the full data regardless of which variables you may choose to use later on.

## Description of the data sets

There are two datasets in this project: players and sessions. While the players data set has one row for each induvidual, the session data set has one row for each played session. One player can be part of several sessions. The data sets has one feature in common: hashedEmail. A hashed email in the session data set can be linked to a player via hashedEmail in the players data set.

Below is more information on each data set.

### Players:
196 observations, 9 variables
- `experience` (chr) - the level of the player
- `subscribe` (lgl) - if the player has subscripted to the game
- `hashedEmail` (chr) - email of the player, hashed to maintain privacy
- `played_hours` (dbl) - amount of hours player has been on the server
- `name` (chr) - player's first name
- `gender` (chr) - Male, Female, Non-binary or Prefer ot to say
- `age` (dbl) - player's age
- `individualId` (lgl) - empty for all examples
- `organizationName` (lgl) - empty for all examples  

#### Issues regarding the players data set

I can not see any directly big issues regarding the data set, but it could be an advantage to modify it.

The last two variables of the data set are unused. Since they do not give any information, they can be filtered away to save storage space, and to make the data set less confusing. Aside from these two columns, the data set does not miss any values, which is good news for a data scientist. I checked this by running the following code which shows an empty data frame.


In [33]:
na_players <- select(players_data, experience:age) |>
    filter(if_any(everything(), is.na))
na_players

experience subscribe hashedEmail played_hours name gender age

The data types also makes sense. At the same time, to make it even more clear what the data we are working with represents, some of the variable types could be changed. Both `played_hours` and `age` are viewed as double types, but all the entites are integeger type. This is probably because double is the default when representing numbers. The same applies for `experience`and `gender` where character is the default type, but these could also be typed as factor, since they consist of a limited amount of categories. Again, this is only suggestions to make the data frame even easier to interpret.

### Sessions:
1535 observations, 5 variables
- `hashedEmail` (chr) - email of the player, hashed to maintain privacy
- `start_time` (chr) - date and time when session started, on format "dd/mm/yyyy 00:00"
- `end_time` (chr) - date and time when session ended, on format "dd/mm/yyyy 00:00"
- `original_start_time` (dbl) - unix timestamp showing the original start time
- `original_end_time` (dbl) - unix timestamp showing the original start time


#### Issues regarding the players data set
sources:
https://bookdown.org/rdpeng/rprogdatascience/dates-and-times.html
https://ubc-dsci.github.io/dsci-100-student/REFERENCE_R.html
https://datasciencebook.ca/reading.html

In [6]:
library(lubridate)
date <- as_datetime("30/06/2024 18:24")
date



Attaching package: ‘lubridate’


The following objects are masked from ‘package:base’:

    date, intersect, setdiff, union




[1] "2030-06-21 00:18:24 UTC"